File: server/pkg/thread/comment.go
package thread

import (
	"time"
)

type Comment struct {
	id      CommentID
	author  UserID
	content string
}

func NewComment(id CommentID, author UserID, content string) *Comment {
	return &Comment{
		id:      id,
		author:  author,
		content: content,
	}
}

func (c *Comment) ID() CommentID {
	return c.id
}

func (c *Comment) Author() UserID {
	return c.author
}

func (c *Comment) Content() string {
	return c.content
}

func (c *Comment) CreatedAt() time.Time {
	return c.id.Timestamp()
}

func (c *Comment) SetContent(content string) {
	c.content = content
}

func (c *Comment) Clone() *Comment {
	if c == nil {
		return nil
	}

	return &Comment{
		id:      c.id.Clone(),
		author:  c.author.Clone(),
		content: c.content,
	}
}

File: web/src/gql/queries/comment.ts
import { gql } from "@apollo/client";

export const ADD_COMMENT = gql`
  mutation AddComment($threadId: ID!, $content: String!) {
    addComment(input: { threadId: $threadId, content: $content }) {
      comment {
        id
        author {
          id
          name
          email
        }
        authorId
        content
        createdAt
      }
    }
  }
`;

export const UPDATE_COMMENT = gql`
  mutation UpdateComment($commentId: ID!, $threadId: ID!, $content: String!) {
    updateComment(input: { commentId: $commentId, threadId: $threadId, content: $content }) {
      comment {
        id
        author {
          id
          name
          email
        }
        authorId
        content
        createdAt
      }
    }
  }
`;

export const DELETE_COMMENT = gql`
  mutation DeleteComment($commentId: ID!, $threadId: ID!) {
    deleteComment(input: { commentId: $commentId, threadId: $threadId }) {
      commentId
    }
  }
`;
